google-ads/src/main/java/com/google/ads/googleads/lib/utils/FieldMasks.java

```diff
@@ -119,6 +124,58 @@ private static void compare(
     }
   }
 
+  /**
+   * Gets the field value of a message from a field mask path.
+   *
+   * @param fieldMaskPath The field mask path.
+   * @param entity The entity to retrieve values from.
+   * @return the field Object.
+   */
+  public static <V> V getFieldValue(String fieldMaskPath, Message entity) {
+    Deque<String> fieldMaskParts = new LinkedList<>(Arrays.asList(fieldMaskPath.split("\\.")));
+    Message currentEntity = entity;
+    while (!fieldMaskParts.isEmpty()) {
+      String fieldMaskPart = fieldMaskParts.remove();
+      Preconditions.checkNotNull(
+          currentEntity, String.format("Cannot get field value. %s is null.", fieldMaskPart));
+      Descriptor descriptor = currentEntity.getDescriptorForType();
+      FieldDescriptor childField = descriptor.findFieldByName(fieldMaskPart);
+      V childValue;
+      try {
+        childValue = (V) currentEntity.getField(childField);
+      } catch (NullPointerException e) {
+        throw new IllegalArgumentException(
```

Comment:
I don't think that makes sense here. While we are parsing the path, it isn't an error because we couldn't parse. The error results from the fact that a user entered an invalid argument that fails to meet some specified condition.

Reply:
Hmm, I'd still prefer not to catch `NullPointerException`. This could be caused by any number of things, not just because the field we passed was null. Can we just allow this block to throw?

Reply:
Looks like there's 2 exception cases we need to handle

- The field doesn't exist (the case this comment is tagged on).
- Attempting to access an element of a repeated/primitive/enum field.

Checking the [style guide](http://go/java-practices/exceptions#checked_unchecked), the checked exception should only be used if both:

- the exception can be avoided by proper use of the API
- and the programmer can take some useful action

I'm actually not sure that either condition holds, so unchecked is actually the right call.

For the first case, if the field doesn't exist, it looks like we can detect this because `getFieldByName()` returns null. In this case, rather than triggering the null pointer exception on `getValue()` it would be better to check if the field is null and throw our own exception. This (1) makes it clear what the failure modes are, (2) avoids catching potentially unrelated exceptions and rewrapping them with the wrong message, and (3) avoids additional cost of throwing 2 exceptions.